### ISSUE TITLE
calculate edges

### ISSUE THREAD

**Issue by Adra8373** [waps101/3DMM_edges#5]
Hi 

Thanks for your code.  I want to ask about computing edge-vertex and edge-face lists in this code 
TR = triangulation(tl,ones(k,1),ones(k,1),ones(k,1));
Ev = TR.edges;
clear TR;
Ef = meshFaceEdges( tl,Ev );

what k is? is it the number of principal component or what ? 

Regards

**Comment by anilbas** [comment]
Yes, k is the number of principal components, such as in the Basel Model, where shapePC is a 3n by k matrix and shapeEV is a k by 1 vector.

**Comment by waps101** [comment]
Actually, in this case k is the number of vertices. k=53490 for the Basel Face Model.

Will

**Comment by Adra8373** [comment]
waps101 that is confused me why it should be number of vertices? I got this could you help me please?

rror using triangulation
At least three input points must be provided to define a triangular mesh.

Error in edegestruct (line 10)
TR = triangulation(tl,ones(k,1),ones(k,1),ones(k,1));

**Comment by waps101** [comment]
triangulation is a built in Matlab function that creates a triangulation object for holding a mesh. We only use it as a way of computing an edge/vertex list so only need the connectivity. The actual vertex positions don't matter, hence we suggested just passing in ones for the x, y and z coordinates. tl should be a m by 3 matrix where m is the number of triangles. k should be the number of vertices in the triangulation. You only need to compute the edge structures if you're not using the Basel Face Model. If you use the BFM, we have already computed the edge structures and provide them in a .mat file.
